python/pyarrow/_dataset_parquet.pyx

```diff
@@ -33,6 +33,12 @@ from pyarrow.includes.libarrow_dataset cimport *
 from pyarrow.includes.libarrow_dataset_parquet cimport *
 from pyarrow._fs cimport FileSystem
 
+IF PARQUET_ENCRYPTION_ENABLED:
```

Comment:
Oh, interesting. I didn't know that Cython allowed compile-time conditionals and that definitely makes things easier here :-)

Reply:
> Oh, interesting. I didn't know that Cython allowed compile-time conditionals and that definitely makes things easier here :-)

Well, they are planning to deprecate it .. (https://github.com/cython/cython/issues/4310). But as long as the discussion isn't yet clear, we can happily use this IMO

Reply:
Well, it's already deprecated apparently.
https://github.com/cython/cython/blob/0f9a74693b6203813ddff0218c77532de307eec0/Cython/Compiler/Parsing.py#L2359-L2364

Reply:
Yes, I think it's fine to use it here, we can look into a replacement later if it becomes actually necessary